beanie/odm/fields.py

```diff
@@ -147,34 +133,70 @@ def _validate(cls, v):
         def __get_pydantic_core_schema__(
             cls, source_type: Type[Any], handler: GetCoreSchemaHandler
         ) -> CoreSchema:
-            return json_or_python_schema(
-                python_schema=no_info_plain_validator_function(cls._validate),
-                json_schema=no_info_plain_validator_function(
-                    cls._validate,
-                    metadata={
-                        "pydantic_js_input_core_schema": str_schema(
-                            pattern="^[0-9a-f]{24}$",
-                            min_length=24,
-                            max_length=24,
-                        )
-                    },
-                ),
-                serialization=plain_serializer_function_ser_schema(
-                    lambda instance: str(instance),
-                    return_schema=str_schema(),
-                    when_used="json",
-                ),
+            definition = core_schema.definition_reference_schema(
+                "PydanticObjectId"
+            )  # used for deduplication
+
+            return core_schema.definitions_schema(
+                definition,
+                [
+                    core_schema.json_or_python_schema(
+                        python_schema=core_schema.no_info_plain_validator_function(
+                            cls._validate
+                        ),
+                        json_schema=core_schema.no_info_after_validator_function(
+                            cls._validate,
+                            core_schema.str_schema(
+                                pattern="^[0-9a-f]{24}$",
+                                min_length=24,
+                                max_length=24,
+                            ),
+                        ),
+                        serialization=core_schema.plain_serializer_function_ser_schema(
+                            str, when_used="json"
```

Comment:
```suggestion
                            lambda instance: str(instance), when_used="json"
```
Please revert back to the original with the lambda expression, due to this method expecting a callable as per Pydantic documentation:
https://docs.pydantic.dev/latest/api/pydantic_core_schema/#pydantic_core.core_schema.plain_serializer_function_ser_schema

Reply:
Okay